cmd/interactive/cmd.go

```diff
@@ -0,0 +1,108 @@
+package interactive
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	tea "github.com/charmbracelet/bubbletea"
+	stellarstation "github.com/infostellarinc/go-stellarstation/api/v1"
+	"github.com/infostellarinc/stellarcli/cmd/util"
+	"github.com/infostellarinc/stellarcli/pkg/apiclient"
+	"github.com/spf13/cobra"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+var (
+	interactiveUse   = util.Normalize("interactive-plan [Satellite ID]")
+	interactiveShort = util.Normalize("Interactive Terminal UI.")
+	interactiveLong  = util.Normalize("Interactive Terminal UI.")
+)
+
+// Create reserve-pass command.
+func NewInteractiveCommand() *cobra.Command {
+	command := &cobra.Command{
+		Use:   interactiveUse,
+		Short: interactiveShort,
+		Long:  interactiveLong,
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 1 {
+				return fmt.Errorf("accepts 1 arg(s), received %d", len(args))
+			}
+
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			debugMode, _ := cmd.PersistentFlags().GetBool("debug")
+
+			log.Printf("Fetching up coming and recent plans for Satellite '%s'.", args[0])
+			conn, err := apiclient.Dial()
+			if err != nil {
+				// don't return an error as it's not a CLI error
+				log.Printf("problem dialing: %v\n", err)
+				return nil
+			}
+
+			client := stellarstation.NewStellarStationServiceClient(conn)
+
+			plansResponse, err := client.ListPlans(cmd.Context(), &stellarstation.ListPlansRequest{
+				SatelliteId: args[0],
+				AosAfter:    timestamppb.New(time.Now().Add(-60 * time.Minute)),
+				AosBefore:   timestamppb.New(time.Now().Add(30 * time.Minute)),
+			})
+			if err != nil {
+				log.Printf("could not retrieve plans for Satellite '%s': %v\n", args[0], err)
+				// don't return an error as it's not a CLI error
+				return nil
+			}
+
+			var selectedPlan *stellarstation.Plan
+
+			for _, plan := range plansResponse.Plan {
+				if plan.Status == stellarstation.Plan_CANCELED {
+					continue
+				}
+				if time.Now().After(plan.GetLosTime().AsTime()) {
```

Comment:
Can interactive mode only operate within aos - los, or should this be OperationTimeEnd?

Reply:
yeah, interactive mode is for live passes.